### ISSUE TITLE
Error while creating subscription using latest SDK (v20.1.0)

### ISSUE THREAD

**Issue by a-patel** [stripe/stripe-dotnet#1352]
I am getting an error while creating a new subscription using SDK version 20.1.0.

Code:

```c#
var subscriptionService = new SubscriptionService("StripeSecretkey");
var subscription = await subscriptionService.CreateAsync(subscriptionOptions); // getting error here

//other try
StripeConfiguration.SetApiKey("StripeSecretkey");
var subscriptionService = new SubscriptionService();
var subscription = await subscriptionService.CreateAsync(subscriptionOptions); // getting error here
```

Getting error in both ways

Error:
One or more errors occurred. (Object reference not set to an instance of an object.)

**Comment by a-patel** [comment]
It is urgent,

**Comment by a-patel** [comment]
FYI: I am able to create a customer using same API key, i.e. Customer service and API key both are working

**Comment by ob-stripe** [comment]
Hi @a-patel, thanks for the report. Do you have a full stacktrace that you could share?

**Comment by a-patel** [comment]
Exception trace:

`   at System.Object.GetType()
   at Stripe.Infrastructure.Middleware.RequestStringBuilder.IsDictionary(Object o)
   at Stripe.Infrastructure.Middleware.RequestStringBuilder.FlattenParamsValue(Object value, String keyPrefix)
   at Stripe.Infrastructure.Middleware.RequestStringBuilder.FlattenParamsDictionary(Dictionary`2 dictionary, String keyPrefix)
   at Stripe.Infrastructure.Middleware.RequestStringBuilder.FlattenParamsValue(Object value, String keyPrefix)
   at Stripe.Infrastructure.Middleware.RequestStringBuilder.FlattenParamsOptions(INestedOptions options, String keyPrefix)
   at Stripe.Infrastructure.Middleware.RequestStringBuilder.CreateQuery(String& requestString, INestedOptions options)
   at Stripe.Infrastructure.ParameterBuilder.ApplyAllParameters[T](Service`1 service, BaseOptions obj, String url, Boolean isListMethod)
   at Stripe.Service`1.PostRequest[T](String url, BaseOptions options, RequestOptions requestOptions)
   at Stripe.SubscriptionService.Create(SubscriptionCreateOptions options, RequestOptions requestOptions)
   at Project.Services.Payment.StripePaymentService.<CreateSubscription>d__9.MoveNext() in Project.Services\Payment\PaymentService.cs:line 91
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at Project.API.Controllers.PaymentController.<Subscribe>d__8.MoveNext() in Project.API\Controllers\PaymentController.cs:line 172`

**Comment by a-patel** [comment]
FYI: I am using ASP.NET Core 2.0

**Comment by ob-stripe** [comment]
Thanks. It looks like the library is tripping when trying to encode your request's parameters. Could you share what goes into `subscriptionOptions`?

**Comment by ob-stripe** [comment]
I was able to reproduce the issue by setting a `null` value inside the `Metadata` dictionary.

This is a bug in the library. We'll release a fix prompty and update here when it's released. Sorry for the inconvenience.

**Comment by a-patel** [comment]
@ob-stripe when will you fix the issue?
it is urgent as I want to test some data.

**Comment by ob-stripe** [comment]
@a-patel Most likely later today. I've already written the fix but I need to get it reviewed by another maintainer before I can release an updated version of the library.

In the meantime, you can clone this repository and pull the `ob-fix-1352` branch if you want to test the fix.

**Comment by a-patel** [comment]
@ob-stripe Thanks a lot. I will wait till EOD.

**Comment by ob-stripe** [comment]
Alternatively, if you replace the null values in your metadata dictionary by empty strings, it should work with the current version and achieve the same result (i.e. it will unset the metadata key).

**Comment by ob-stripe** [comment]
Fixed in 20.2.0.
